lib/matplotlib/patches.py

```diff
@@ -416,14 +425,35 @@ def set_color(self, c):
         Patch.set_facecolor, Patch.set_edgecolor
             For setting the edge or face color individually.
         """
-        self.set_facecolor(c)
         self.set_edgecolor(c)
+        self.set_hatchcolor(c)
+        self.set_facecolor(c)
+
+    def _set_hatchcolor(self, color):
+        color = mpl._val_or_rc(color, 'hatch.color')
+        if color in ('inherit', 'edge'):
```

Comment:
It seems 'inherit' and 'edge' are a bit mixed up. In code and documentation. We do not need 'inherit' when we support 'edge'. Please also check the rest of the PR, e.g. "what's new" and the example also still use 'inherit'.

Reply:
Do you mean that the rcParam of `hatch.color` should be changed to `edge` instead of `inherit`? In this case, there might be a problem in `rcsetup.py`, as currently I am using `validate_color_or_inherit` but there is no equivalent for `edge`.

Reply:
We currently have:
- "inherit" is supported on
  - rcParams "legend.facecolor" / "legend.edgecolor" - meaning use "axes.facecolor" / "axes.edgecolor"
  - rcParams "x/ytick.labelcolor" - meaning use "x/ytick.color"
- `Collections.set_edgecolor` supports "face" to set the edgecolor to the facecolor. There's a bit of convoluted logic (which is another topic), but it evenentually resolves to rcParam "patch.edgecolor", which currently only `validate_color`; i.e. you currently cannot set the collections edgecolor to "face" via rcParams.

"inherit" is currently used for taking the "same" color-property (edge/face) from a related Artist. The hatch case is more like the Collections case: take the color from a different color-property of the same artist. I therefore think that naming `hatchcolor="edge"` / `rcParams["hatch.color"] = "edge"` is more precise and consistent than using "inherit". Consequently, we should introduce a new `validate_color_or_edge` validator for "hatch.color".